src/main/scala/ch/epfl/bluebrain/nexus/kg/async/DistributedCache.scala

```diff
@@ -341,27 +344,39 @@ object DistributedCache {
     /**
       * @return true if the project ref is already present so that we can chain the call during an update
       */
-    private def addProjectToAccount(ref: ProjectRef,
-                                    accountRef: AccountRef,
-                                    instant: Instant,
-                                    updateRev: Boolean): Future[Boolean] = {
+    private def addProjectToAccount(ref: ProjectRef, accountRef: AccountRef, updateRev: Boolean): Future[Boolean] = {
       projects(accountRef).flatMap { projects =>
         if (projects.contains(ref)) Future.successful(updateRev)
         else {
-          val empty  = LWWRegister(TimestampedValue(0L, Set.empty[ProjectRef]))
-          val value  = TimestampedValue(instant.toEpochMilli, projects + ref)
-          val update = Update(accountProjectsKey(accountRef), empty, WriteMajority(tm.duration))(_.withValue(value))
+          val empty = LWWRegister(RevisionValue(0L, Set.empty[ProjectRef]))
+          val update = Update(accountProjectsKey(accountRef), empty, WriteMajority(tm.duration)) { currentState =>
+            val currentRevision = currentState.value.rev
+            val currentValue    = currentState.value.value
+            currentValue.find(_.id == ref.id) match {
+              case Some(r) =>
+                currentState.withValue(RevisionValue(currentRevision + 1, currentValue - r + ref))
```

Comment:
Relying on a simple counter is dangerous when performing writes from multiple nodes as it won't converge in concurrent modifications. Please create a ticket for a better implementation based on ORSets or custom data type.